wordpress-mobile/WordPress-iOS
# Stock photos: pagination

Implements part of #8954 

In this PR:

Requests to the stock photos endpoint are paginated. I could not find any documentation in the [WordPress REST API docs](https://developer.wordpress.org/rest-api/) related to the endpoint we use (`/rest/v1/meta/external-media/pexels`) but I looked at what [other media-related endpoints](https://developer.wordpress.org/rest-api/reference/media/) do to paginate requests.

Implementation details:
- In a similar fashion to what Spring does, I added two interfaces (`ResultsPage` and `Pageable`) that encapsulated, in that order, the paginated results and the pagination parameters.
- The logic to decide when a new page of data needs to be requested is in `StockPhotosDataSource` which, imo, is less than ideal, but on the other hand, I did not see any other way to know when the collection view is reaching the end of its scroll area.

To test:
- Search for a keyword with only one page of results (i.e. Pizza) and scroll down. Once the bottom is reached, there should not be any other requests.
- Search for a keyword with several pages of results (i.e. Car), and scroll down. Before reaching the bottom of the results collection view, a new batch of images should be loaded.

Next steps: Tracks, cleanup and some unit tests
